In [1]:
%matplotlib inline
import pandas as pd
import matplotlib.pyplot as plt

import numpy as np
import datetime as dt

In [2]:
########################################################################################################

In [3]:
raw_measurements_df = pd.read_csv('hawaii_measurements.csv')
raw_stations_df = pd.read_csv('hawaii_stations.csv')

In [4]:
raw_measurements_df.tail()

station        date  prcp  tobs
19545  USC00516128  2017-08-19  0.09    71
19546  USC00516128  2017-08-20   NaN    78
19547  USC00516128  2017-08-21  0.56    76
19548  USC00516128  2017-08-22  0.50    76
19549  USC00516128  2017-08-23  0.45    76

In [5]:
raw_measurements_df.dtypes

station     object
date        object
prcp       float64
tobs         int64
dtype: object

In [6]:
#Find duplicate rows
search = pd.DataFrame.duplicated(raw_measurements_df)
search[search == True]

Series([], dtype: bool)

No duplicate rows found

Look for missing data

In [7]:
raw_measurements_df.shape

(19550, 4)

In [8]:
raw_measurements_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19550 entries, 0 to 19549
Data columns (total 4 columns):
station    19550 non-null object
date       19550 non-null object
prcp       18103 non-null float64
tobs       19550 non-null int64
dtypes: float64(1), int64(1), object(2)
memory usage: 611.0+ KB


In [9]:
raw_measurements_df.describe()

prcp          tobs
count  18103.000000  19550.000000
mean       0.160644     73.097954
std        0.468746      4.523527
min        0.000000     53.000000
25%        0.000000     70.000000
50%        0.010000     73.000000
75%        0.110000     76.000000
max       11.530000     87.000000

There is data missing in the raw measurements dataframe. prcp has only 18103 values 

In [10]:
#Drop rows with NaN or null values
clean_raw_measurements_df= raw_measurements_df.dropna()
clean_raw_measurements_df

station        date  prcp  tobs
0      USC00519397  2010-01-01  0.08    65
1      USC00519397  2010-01-02  0.00    63
2      USC00519397  2010-01-03  0.00    74
3      USC00519397  2010-01-04  0.00    76
5      USC00519397  2010-01-07  0.06    70
6      USC00519397  2010-01-08  0.00    64
7      USC00519397  2010-01-09  0.00    68
8      USC00519397  2010-01-10  0.00    73
9      USC00519397  2010-01-11  0.01    64
10     USC00519397  2010-01-12  0.00    61
11     USC00519397  2010-01-14  0.00    66
12     USC00519397  2010-01-15  0.00    65
13     USC00519397  2010-01-16  0.00    68
14     USC00519397  2010-01-17  0.00    64
15     USC00519397  2010-01-18  0.00    72
16     USC00519397  2010-01-19  0.00    66
17     USC00519397  2010-01-20  0.00    66
18     USC00519397  2010-01-21  0.00    69
19     USC00519397  2010-01-22  0.00    67
20     USC00519397  2010-01-23  0.00    67
21     USC00519397  2010-01-24  0.01    71
22     USC00519397  2010-01-25  0.00    67
23     USC00519397  2010-01-26  0.04    76
24     USC00519397  2010-01-27  0.12    68
25     USC00519397  2010-01-28  0.00    72
27     USC00519397  2010-01-31  0.03    67
28     USC00519397  2010-02-01  0.01    66
30     USC00519397  2010-02-04  0.01    69
31     USC00519397  2010-02-05  0.00    67
32     USC00519397  2010-02-06  0.00    67
...            ...         ...   ...   ...
19513  USC00516128  2017-07-17  0.39    72
19514  USC00516128  2017-07-18  2.40    77
19515  USC00516128  2017-07-19  0.27    74
19516  USC00516128  2017-07-20  0.70    75
19517  USC00516128  2017-07-21  0.10    72
19518  USC00516128  2017-07-22  4.00    72
19519  USC00516128  2017-07-23  0.80    78
19520  USC00516128  2017-07-24  0.84    77
19521  USC00516128  2017-07-25  0.30    79
19522  USC00516128  2017-07-26  0.30    73
19523  USC00516128  2017-07-27  0.00    75
19524  USC00516128  2017-07-28  0.40    73
19525  USC00516128  2017-07-29  0.30    77
19526  USC00516128  2017-07-30  0.30    79
19527  USC00516128  2017-07-31  0.00    74
19529  USC00516128  2017-08-02  0.25    80
19530  USC00516128  2017-08-03  0.06    76
19533  USC00516128  2017-08-07  0.05    78
19534  USC00516128  2017-08-08  0.34    74
19535  USC00516128  2017-08-09  0.15    71
19536  USC00516128  2017-08-10  0.07    75
19538  USC00516128  2017-08-12  0.14    74
19540  USC00516128  2017-08-14  0.22    79
19541  USC00516128  2017-08-15  0.42    70
19542  USC00516128  2017-08-16  0.42    71
19543  USC00516128  2017-08-17  0.13    72
19545  USC00516128  2017-08-19  0.09    71
19547  USC00516128  2017-08-21  0.56    76
19548  USC00516128  2017-08-22  0.50    76
19549  USC00516128  2017-08-23  0.45    76

[18103 rows x 4 columns]

In [11]:
clean_raw_measurements_df.shape

(18103, 4)

In [12]:
#Find duplicate rows
search = pd.DataFrame.duplicated(raw_stations_df)
search[search == True]

Series([], dtype: bool)

No duplicate rows found
Look for missing data

In [13]:
raw_stations_df.shape

(9, 5)

In [14]:
raw_stations_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9 entries, 0 to 8
Data columns (total 5 columns):
station      9 non-null object
name         9 non-null object
latitude     9 non-null float64
longitude    9 non-null float64
elevation    9 non-null float64
dtypes: float64(3), object(2)
memory usage: 440.0+ bytes


In [15]:
raw_stations_df.describe()

latitude   longitude   elevation
count   9.000000    9.000000    9.000000
mean   21.393826 -157.867098   60.977778
std     0.086442    0.103873  103.465547
min    21.271600 -158.011100    0.900000
25%    21.333100 -157.975100    7.000000
50%    21.393400 -157.837400   14.600000
75%    21.451670 -157.802500   32.900000
max    21.521300 -157.711390  306.600000

Stations Data looks clean

In [16]:
#Write clean data to new CSV files
clean_raw_measurements_df.to_csv('clean_hawaii_measurements.csv')
raw_stations_df.to_csv('clean_hawaii_stations.csv')